JitsiMeetJS.js

```diff
@@ -69,28 +63,28 @@ function getLowerResolution(resolution) {
 }
 
 /**
- * Checks the available devices in options and concatenate the data to the
- * name, which will be used as analytics event name. Adds resolution for the
- * devices.
- * @param name name of event
- * @param options gum options
- * @returns {*}
+ * Extracts from an 'options' objects with a specific format
+ * (TODO what IS the format?) the attributes which are to be logged in analytics
+ * events.
+ *
+ * @param options gum options (???)
+ * @returns {*} the attributes to attach to analytics events.
  */
-function addDeviceTypeToAnalyticsEvent(name, options) {
-    let ret = name;
-
-    if (options.devices.indexOf('audio') !== -1) {
-        ret += '.audio';
-    }
-    if (options.devices.indexOf('desktop') !== -1) {
-        ret += '.desktop';
-    }
-    if (options.devices.indexOf('video') !== -1) {
-        // we have video add resolution
-        ret += `.video.${options.resolution}`;
+function getAnalyticsAttributesFromOptions(options) {
+    const attributes = {
+        audioRequested:
+            options.devices.indexOf('audio') !== -1,
+        videoRequested:
+            options.devices.indexOf('video') !== -1,
+        screenSharingRequested:
+            options.devices.indexOf('desktop') !== -1
```

Comment:
You could get away with using Array.prototype.includes here if desired instead of indexOf. The method gets polyfilled by something somewhere.

Reply:
Sure